techlibs/xilinx/synth_xilinx.cc

```diff
@@ -235,32 +253,28 @@ struct SynthXilinxPass : public Pass
 			Pass::call(design, "dffsr2dff");
 			Pass::call(design, "dff2dffe");
 			Pass::call(design, "opt -full");
-			if (vpr)
-			{
+
+			 if (vpr) {
 				Pass::call(design, "techmap -map +/techmap.v -map +/xilinx/arith_map.v -D _EXPLICIT_CARRY");
-				Pass::call(design, "hierarchy -check");
-			}
-			else
-			{
-				Pass::call(design, "techmap -map +/techmap.v -map +/xilinx/arith_map.v");
+			} else {
+				Pass::call(design, "techmap -map +/techmap.v -map +/xilinx/arith_map.v -map +/xilinx/ff_map.v");
```

Comment:
Why have ff map here?

Reply:
It has to be before the `abc` pass. When there is a FF with inverter on eg. CLR signal then it is added as Yosys' internal `$not` gate. That gate must be converted to LUT during the `abc` pass. If I do FF mapping after ABC then I am left with unmapped `$not` gates.

Reply:
There should never be an invert on the CLR signal?  For example, a FDCE (Async clear) and FDPE (Async set) should be used to handle inverts on the CLR singal.

Reply:
I guess this is an inverter on when to do the clear?  Weird.